Frontend/Gmfctn/src/app/shared/components/profile-setting-modal/profile-setting-modal.component.ts

```diff
@@ -29,10 +33,24 @@ export class ProfileSettingModalComponent implements OnInit {
     });
   }
 
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
+
   updateUser(): void {
     if (this.userForm.valid) {
-      this.user = {...this.user, ...this.userForm.value }
-      this.dialogRef.close(this.user);
+      this.user = {...this.user, ...this.userForm.value };
+      const status: string = this.userForm.value.status;
+      const data = {
+        email: this.user.email,
+        firstName: this.user.firstName,
+        lastName: this.user.lastName,
+        userName: this.user.userName,
+        status: status
+      };
+
+      this.subscription.add(this.profileService.updateUser(data).subscribe());
```

Comment:
to avoid adding a single request to subscriptions you could use `.pipe(take(1))` or `.pipe(first())`. Both of this operators are unsubscribing from observable after value received

Reply:
Fixed